compiler/src/iree/compiler/Dialect/LinalgExt/IR/LinalgExtOps.td

```diff
@@ -1570,4 +1570,100 @@ def IREELinalgExt_WinogradOutputTransformOp : IREELinalgExt_Op<"winograd.output_
 
 } // OpGroupWinogradOps
 
+//===---------------------------------------------------------------------===//
+// Custom tilable op
+//===---------------------------------------------------------------------===//
+
+def IREELinalgExt_CustomOp : IREELinalgExt_Op<"custom_op"> {
+  let summary = "Custom operation for compiling with IREE";
+  let description = [{
+    This operation is meant to allow computation sequences that are fused at
+    tile level prescriptively. This is to account for cases where such fusion
+    cannot/is not yet discovered appropriately.
+
+    The operation implements all the interfaces needed to be able to
+    1. Compile e2e using IREE
+    2. Still be able to fuse with other operations that the compiler can
+       figure out automatically.
+
+    Similar to how `LinalgOp`s represent a perfectly nested loop computation
+    with
+    - `indexing_maps` representing how the `ins`/`outs` are accessed
+    - `region` representing the scalar computation performed
+    - `iterator_types` representing the dependence along each iteration space
+      dimension
+    this operation represent a tiled computation with perfectly nested
+    inter-tile loop nest.
+    - `indexing_maps` represent what slices slices of the `ins`/`outs` are
+      needed for each iteration of the tiled computation.
+    - `region` represents the tiled computation performed using these slices
+    - `iterator_types` represents the dependence between tiles along each
+      iteration space.
+
+    Some modifications required to handle the tile-level semantics are
+    - Some dimensions of operands might not be accessed by dimensions of the
+      inter-tile iteration space. This means that along these dimensions the
+      slice size matches the dimension size. This access pattern of operands
+      is captured in the respective indexing map using a `symbol` to represent
+      that the entire dimension needs to be sliced.
```

Comment:
What would a `<constant>`, `d0 + <constant>`, or `s0 + <constant>` mean in one of the indexing_maps with this? Should we disallow such cases in the verifier?

Reply:
Good point... I think that is TBD based on what the entire codegen looks like. Will need to back-port some restrictions after have everything working end-to-end.